inst/include/child_model_simulation.hpp

```diff
@@ -13,54 +12,57 @@ void run_child_ageing(int t,
                       const State<ModelVariant, real_type> &state_curr,
                       State<ModelVariant, real_type> &state_next,
                       IntermediateData<ModelVariant, real_type> &intermediate) {
-  const auto demog = pars.base.demography;
-  const auto cpars = pars.children.children;
   static_assert(ModelVariant::run_child_model,
                 "run_hiv_child_infections can only be called for model variants where run_child_model is true");
-  constexpr auto ss = StateSpace<ModelVariant>().base;
-  constexpr auto hc_ss = StateSpace<ModelVariant>().children;
-
-  for (int s = 0; s < ss.NS; ++s) {
-    //less than 5 because there is a cd4 transition between ages 4 and 5
-    for (int a = 1; a < hc_ss.hc2_agestart; ++a) {
-      for (int hd = 0; hd < hc_ss.hc1DS; ++hd) {
-        for (int cat = 0; cat < hc_ss.hcTT; ++cat) {
-          state_next.children.hc1_hiv_pop(hd, cat, a, s) +=
-            state_curr.children.hc1_hiv_pop(hd, cat, a - 1, s) * demog.survival_probability(a, s, t);
+  constexpr auto ss_b = StateSpace<ModelVariant>().base;
+  constexpr auto ss_c = StateSpace<ModelVariant>().children;
+  const auto& p_dm = pars.base.demography;
+  const auto& p_hc = pars.children.children;
+  const auto& p_op = pars.base.options;
+  const auto& c_hc = state_curr.children;
+  auto& n_hc = state_next.children;
+
+  for (int s = 0; s < ss_b.NS; ++s) {
+    // less than 5 because there is a cd4 transition between ages 4 and 5
+    for (int a = 1; a < ss_c.hc2_agestart; ++a) {
+      for (int hd = 0; hd < ss_c.hc1DS; ++hd) {
+        for (int cat = 0; cat < ss_c.hcTT; ++cat) {
+          n_hc.hc1_hiv_pop(hd, cat, a, s) += c_hc.hc1_hiv_pop(hd, cat, a - 1, s) * p_dm.survival_probability(a, s, t);
         }
-        for (int dur = 0; dur < ss.hTS; ++dur) {
-          state_next.children.hc1_art_pop(dur, hd, a, s) +=
-            state_curr.children.hc1_art_pop(dur, hd, a - 1, s) * demog.survival_probability(a, s, t);
+        for (int dur = 0; dur < ss_b.hTS; ++dur) {
+          n_hc.hc1_art_pop(dur, hd, a, s) += c_hc.hc1_art_pop(dur, hd, a - 1, s) * p_dm.survival_probability(a, s, t);
         }
       }
     }
   }
 
-  for (int s = 0; s < ss.NS; ++s) {
-    for (int hd = 0; hd < hc_ss.hc1DS; ++hd) {
-      for (int hd_alt = 0; hd_alt < hc_ss.hc2DS; ++hd_alt) {
-        for (int cat = 0; cat < hc_ss.hcTT; ++cat) {
-          state_next.children.hc2_hiv_pop(hd_alt, cat, 0, s) += state_curr.children.hc1_hiv_pop(hd, cat, hc_ss.hc1_ageend, s) * demog.survival_probability(hc_ss.hc2_agestart, s, t) * cpars.hc_cd4_transition(hd_alt, hd);
+  for (int s = 0; s < ss_b.NS; ++s) {
+    for (int hd = 0; hd < ss_c.hc1DS; ++hd) {
+      for (int hd_alt = 0; hd_alt < ss_c.hc2DS; ++hd_alt) {
+        const auto cd4_transition_prob_per_group = p_dm.survival_probability(ss_c.hc2_agestart, s, t) *
+                                                   p_hc.hc_cd4_transition(hd_alt, hd);
```

Comment:
common variable extracted

Reply:
Conceptually I like this, but here it doesn't make much sense because we are representing two different processes: natural mortality and transition from u5 CD4 categories to o5 CD4 categories. We can add this in some other places though (initial thought being in the fertility calculations)

Reply:
marking for call tomorrow